test/acceptance/features/immutableBindings.feature

```diff
@@ -0,0 +1,192 @@
+            metadata:
+                name: binding-immutable
+            spec:
+                application:
+                    name: app-immutable-2
+                    group: apps
+                    version: v1
+                    resource: deployments
+                services:
+                  - group: stable.example.com
+                    version: v1
+                    kind: Backend
+                    name: service-immutable
+            """
+
+    Scenario: Allow modifying a Service Binding when it is not yet ready due to "Application Not Found"
+        And Service Binding is applied
+            """
+            apiVersion: binding.operators.coreos.com/v1alpha1
+            kind: ServiceBinding
+            metadata:
+                name: binding-immutable-2
+            spec:
+                services:
+                  - group: stable.example.com
+                    version: v1
+                    kind: Backend
+                    name: service-immutable
+                application:
+                    name: app1
+                    group: apps
+                    version: v1
+                    resource: deployments
+            """
+        And jq ".status.conditions[] | select(.type=="Ready").status" of Service Binding "binding-immutable-2" should be changed to "False"
+        When Generic test application "app2" is running
+        And Service Binding is applied
+            """
+            apiVersion: binding.operators.coreos.com/v1alpha1
+            kind: ServiceBinding
+            metadata:
+                name: binding-immutable-2
+            spec:
+                services:
+                  - group: stable.example.com
+                    version: v1
+                    kind: Backend
+                    name: service-immutable
+                application:
+                    name: app2
+                    group: apps
+                    version: v1
+                    resource: deployments
+            """
+        Then Service Binding "binding-immutable-2" is ready
+
+
+    Scenario: Allow modifying a Service Binding when it is not yet ready due to "Service Not Found"
+        Given Generic test application "app3" is running
+        And Service Binding is applied
+            """
+            apiVersion: binding.operators.coreos.com/v1alpha1
+            kind: ServiceBinding
+            metadata:
+                name: binding-immutable-3
+            spec:
+                services:
+                  - group: stable.example.com
+                    version: v1
+                    kind: Backend
+                    name: service2
+                application:
+                    name: app3
+                    group: apps
+                    version: v1
+                    resource: deployments
+            """
+        And jq ".status.conditions[] | select(.type=="Ready").status" of Service Binding "binding-immutable-3" should be changed to "False"
```

Comment:
A step for verifying the actual reason is "Service Not Found" defined in the scenario should be added here so that the test fails propely on any other reason then expected.

Reply:
see above.